### PR TITLE
Add a few initial proposal ideas, more to come

### PR DESCRIPTION
**Pull request by ktoso**
A few initial project ideas while we still gather ideas and potential mentors, so that the page isn't completely empty while the application goes through :)

### CODE REVIEW

File: gsoc2023/index.md [hunk 1]
```diff
@@ -20,4 +20,76 @@ If you would like to reach out to a mentor privately, rather than making a publi
...
+
+- [Egor Zhdan](https://github.com/egorzhdan)
+
+### Swift Package Manager
+
+### Scripting in Swift
+
+**Project size**: Medium
+
+*Recommended skills*
```
Comment (ktoso) [inline gsoc2023/index.md:56]:
```suggestion
**Recommended skills**
```

File: gsoc2023/index.md [hunk 2]
```diff
@@ -20,4 +20,76 @@ If you would like to reach out to a mentor privately, rather than making a publi
...
+
+There's a pitch (https://forums.swift.org/t/pitch-swiftpm-support-for-swift-scripts-revision/46717) which outlines an initial design from a previous GSoC proposal that could be used as the basis of this work. It may need some polish, figuring out the details the project could aim for implementing a basic version of it.
+
+**Expected outcomes/benefits/deliverables**
+
+The goal of this project is to implement basic support for defining package dependencies for use in a script.
+
+**Potential mentors**
+
+* [Boris Bügling](https://github.com/neonichu)
```
Comment (ktoso) [inline gsoc2023/index.md:75]:
```suggestion
- [Boris Bügling](https://github.com/neonichu)
```

File: gsoc2023/index.md [hunk 3]
```diff
@@ -20,4 +20,76 @@ If you would like to reach out to a mentor privately, rather than making a publi
...
+**Description**
+SwiftPM currently supports a handful of hardcoded templates that can act as a starting point for user projects. Since there are many different needs for such templates, there should be a way to customize these templates and share them with others without having to make changes to the SwiftPM codebase itself. It could also be useful to make templates interactive, so e.g. users can pick between including tests by default or choose a set of dependencies to include. In this project, we will define a format for templates, design an API for making them interactive and integrate these into the SwiftPM commandline tool in a user-friendly way.
+
+**Expected outcomes/benefits/deliverables**
+
+- The goal of this project is to implement support for customizable package templates in the SwiftPM commandline tool.
+
+**Potential mentors**
+
+* [Boris Bügling](https://github.com/neonichu)
```
Comment (ktoso) [inline gsoc2023/index.md:94]:
```suggestion
- [Boris Bügling](https://github.com/neonichu)
```

**Comment by ktoso** [comment]
@swift-ci please test

**Comment by ktoso** [comment]
This can be merged at any time, thanks in advance for review :)

**Review by shahmishal — APPROVED** [review]

**Pull request merged**